circuit_knitting/cutting/cutting_experiments.py

```diff
@@ -0,0 +1,75 @@
+# This code is a Qiskit project.
+
+# (C) Copyright IBM 2023.
+
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Functions for evaluating circuit cutting experiments."""
+
+from __future__ import annotations
+
+from qiskit.circuit import QuantumCircuit
+from qiskit.quantum_info import PauliList
+
+from .qpd import WeightType
+from .cutting_evaluation import _generate_cutting_experiments
+
+
+def generate_cutting_experiments(
+    circuits: QuantumCircuit | dict[str | int, QuantumCircuit],
+    observables: PauliList | dict[str | int, PauliList],
+    num_samples: int | float,
+) -> tuple[
+    list[QuantumCircuit] | dict[str | int, list[QuantumCircuit]],
+    list[tuple[float, WeightType]],
+]:
+    r"""
+    Generate cutting subexperiments and their associated weights.
+
+    If the input, ``circuits``, is a :class:`QuantumCircuit` instance, the
+    output subexperiments will be contained within a 1D array, and ``observables`` is
+    expected to be a :class:`PauliList` instance.
+
+    If the input circuit and observables are specified by dictionaries with partition labels
+    as keys, the output subexperiments will be returned as a dictionary which maps each
+    partition label to a 1D array containing the subexperiments associated with that partition.
+
+    In both cases, the subexperiment lists are ordered as follows:
+
+        :math:`[sample_{0}observable_{0}, \ldots, sample_{0}observable_{N}, \ldots, sample_{M}observable_{N}]`
```

Comment:
This is a rehash of my comment at https://github.com/Qiskit-Extensions/circuit-knitting-toolbox/pull/385#discussion_r1311805133

```suggestion
        :math:`[sample_{0}observable_{0}, \ldots, sample_{0}observable_{N}, sample_{1}observable_{0}, \ldots, sample_{M}observable_{N}]`
```

Not sure if you chose to decline or just didn't notice the one remaining change I suggested.

Reply:
> This is a rehash of my comment at [#385 (comment)](https://github.com/Qiskit-Extensions/circuit-knitting-toolbox/pull/385#discussion_r1311805133)
> 
> Not sure if you chose to decline or just didn't notice the one remaining change I suggested.

I guess it seemed asymmetric to show one indexing from [0,N] and the other indexing from [0,1,M]. I would either include the 1 in each case (I think that was my original way) or remove it in each case (what I have here). What do you think?

Reply:
The reason I chose to do it that way is to show explicitly that (1,0) comes immediately after (0, N).  The way you have written it, (0, N) could (in principle) appear _anywhere_ in the list.

Reply:
> The way you have written it, (0, N) could (in principle) appear anywhere in the list.

To follow up, the way you wrote it originally did not have this problem, only the way it appears currently.